internal/domain/file_content_response.go

```diff
@@ -0,0 +1,9 @@
+package domain
+
+type FileContentResponse struct {
+	Filename  string `json:"filename"`
+	Content   string `json:"content"`
+	StartLine int    `json:"start_line"`
```

Comment:
It may be prudent to include a custom method to calculate or validate the Position value based on StartLine and EndLine. This helps ensure that Position corresponds correctly to the range defined by StartLine and EndLine, enhancing the logical consistency of your responses.